cdap-watchdog/src/main/java/co/cask/cdap/logging/read/AvroFileReader.java

```diff
@@ -181,9 +171,53 @@ public void readLog(Location file, Filter logFilter, long fromTimeMs, long toTim
     }
   }
 
+  /**
+   *  Read current block in Avro file from current block sync marker to next block sync marker
+   */
+
+  private List<LogEvent> readToEndSyncPosition(DataFileReader<GenericRecord> dataFileReader, Filter logFilter,
+                                               long fromTimeMs, long endSyncPosition) throws IOException {
+
+    List<LogEvent> logSegment = new ArrayList<>();
+    GenericRecord datum = null;
+    long currentSyncPosition = dataFileReader.previousSync();
+    // Read till the end if endSyncPosition is not known (in case of open file)
+    // or read until endSyncPosition has reached
+    while (dataFileReader.hasNext() && (endSyncPosition == -1 || (currentSyncPosition < endSyncPosition))) {
+      datum = dataFileReader.next(datum);
+      ILoggingEvent loggingEvent = LoggingEvent.decode(datum);
+
+      // Stop when reached fromTimeMs
+      if (loggingEvent.getTimeStamp() > fromTimeMs) {
+        break;
+      }
+
+      if (logFilter.match(loggingEvent)) {
+        logSegment.add(new LogEvent(loggingEvent,
+                                    new LogOffset(LogOffset.INVALID_KAFKA_OFFSET, loggingEvent.getTimeStamp())));
+      }
+      currentSyncPosition = dataFileReader.previousSync();
+    }
+
+    return logSegment;
+  }
+
+  private long skipToPosition(DataFileReader<GenericRecord> dataFileReader,
```

Comment:
It would be good to have javadoc that explains what this method is trying to do.